### PR TITLE
Fix panic warning and add arbitrary Duration

### PR DESCRIPTION
**Pull request by celinval**
Add an Arbitrary implementation for Duration and also fix unused variable warnings when panic message uses a variable in its format string.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: library/kani/src/arbitrary.rs [hunk 1]
```diff
@@ -170,3 +170,11 @@ where
         Box::new(T::any())
     }
 }
+
+impl Arbitrary for std::time::Duration {
+    fn any() -> Self {
+        const NANOS_PER_SEC: u32 = 1_000_000_000;
+        let nanos = u32::any() % NANOS_PER_SEC;
```
Comment (zhassan-aws) [inline library/kani/src/arbitrary.rs:177]: How does this perform compared to using an assumption (`nanos < NANOS_PER_SEC`)?

File: tests/ui/should-panic-attribute/expected-panics/test.rs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright Kani Contributors
 // SPDX-License-Identifier: Apache-2.0 OR MIT
-
+//
+// compile-flags: -Copt-level=1
```
Comment (zhassan-aws) [inline tests/ui/should-panic-attribute/expected-panics/test.rs:4]: Why was this added?

Comment (celinval) [reply tests/ui/should-panic-attribute/expected-panics/test.rs:4]: Because the expected file checks the number of existing assertions which went up from 2 to 6. However, those are dead assumption.

File: library/kani/src/arbitrary.rs (hunk 1 above)
Comment (celinval) [reply library/kani/src/arbitrary.rs:177]: great question... I didn't spend time optimizing this implementation since my use case was fairly simple.

Comment (zhassan-aws) [reply library/kani/src/arbitrary.rs:177]: I see. I'm worried that a modulo operation with a number that is not 2^n might be expensive.

Comment (celinval) [reply library/kani/src/arbitrary.rs]: I added a performance test. The original version took around 14s and the assume one took close to 12s. So not a huge difference, but assume seems better. I also tried using a conditional assignment instead of a global assumption, but there wasn't any difference, so I'm keeping it simple for now and using the assume version.

**Review by zhassan-aws — APPROVED** [review]
